src/image_bios.cpp

```diff
@@ -0,0 +1,196 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright (C) 2021 YADRO.
+ */
+
+#include "config.h"
+
+#include "image_bios.hpp"
+
+#include "dbus.hpp"
+#include "fwupderr.hpp"
+#include "subprocess.hpp"
+#include "tracer.hpp"
+
+#include <chrono>
+#include <fstream>
+#include <thread>
+
+static const fs::path aspeedScm = "/sys/bus/platform/drivers/aspeed-smc";
+static constexpr auto spiDriver = "1e631000.spi";
+static constexpr auto gpioNamePCHPower = "PWRGD_DSW_PWROK";
+static constexpr auto gpioNameBIOSSel = "MIO_BIOS_SEL";
```

Comment:
Looks like this may be not enough. See the internal PR.

Reply:
It seems to me that it is enough. I wrote the detailed answer in the internal PR.